pkg/controller/jobs/rayjob/rayjob_controller.go

```diff
@@ -113,12 +113,13 @@ func (j *RayJob) PodSets() []kueue.PodSet {
 }
 
 func (j *RayJob) RunWithPodSetsInfo(podSetInfos []jobframework.PodSetInfo) error {
-	j.Spec.Suspend = false
 	expectedLen := len(j.Spec.RayClusterSpec.WorkerGroupSpecs) + 1
 	if len(podSetInfos) != expectedLen {
 		return jobframework.BadPodSetsInfoLenError(expectedLen, len(podSetInfos))
 	}
 
+	j.Spec.Suspend = false
```

Comment:
Is this a fix for some rare scenario discovered  by the test changes, or it is transparent?

Reply:
It is more  or less a fix, without this , even if the info is not acceptable the  `j.Spec.Suspend` wold have been changed.

In the full implementation, it wold not have any effect since the change is not pushed to the API, but the behavior wold have been strange in the tests.

Reply:
sgtm